mcs/restapi/admin_config.go

```diff
@@ -0,0 +1,133 @@
+		}
+		return admin_api.NewListConfigOK().WithPayload(configListResp)
+	})
+	// Configuration Info
+	api.AdminAPIConfigInfoHandler = admin_api.ConfigInfoHandlerFunc(func(params admin_api.ConfigInfoParams) middleware.Responder {
+		config, err := getConfigResponse(params)
+		if err != nil {
+			return admin_api.NewConfigInfoDefault(500).WithPayload(&models.Error{Code: 500, Message: swag.String(err.Error())})
+		}
+		return admin_api.NewConfigInfoOK().WithPayload(config)
+	})
+
+}
+
+// listConfig gets all configurations' names and their descriptions
+func listConfig(client MinioAdmin) (madmin.Help, error) {
+	hr, err := client.helpConfigKV("", "", false)
+	if err != nil {
+		return madmin.Help{}, err
+	}
+	return hr, nil
+}
+
+// getListConfigResponse performs listConfig() and serializes it to the handler's output
+func getListConfigResponse() (*models.ListConfigResponse, error) {
+	mAdmin, err := newMAdminClient()
+	if err != nil {
+		log.Println("error creating Madmin Client:", err)
+		return nil, err
+	}
+	// create a MinIO Admin Client interface implementation
+	// defining the client to be used
+	adminClient := adminClient{client: mAdmin}
+
+	configKeys, err := listConfig(adminClient)
+	if err != nil {
+		log.Println("error listing configurations:", err)
+		return nil, err
+	}
+	// serialize output
+	var configDescs []*models.ConfigDescription
+	for _, c := range configKeys.KeysHelp {
+		desc := &models.ConfigDescription{
+			Key:         c.Key,
+			Description: c.Description,
+		}
+		configDescs = append(configDescs, desc)
+	}
+	listGroupsResponse := &models.ListConfigResponse{
+		Configurations: configDescs,
+	}
+	return listGroupsResponse, nil
+}
+
+// getConfig gets the key values for a defined configuration
+func getConfig(client MinioAdmin, name string) (madmin.Target, error) {
+	config, err := client.getConfigKV(name)
+	if err != nil {
+		return madmin.Target{}, err
+	}
+	// config comes as an array []madmin.Target
+	if len(config) > 0 {
+		// return Key Values, first element contains info
+		return config[0], nil
+	}
+	return madmin.Target{}, nil
+}
+
+// getConfigResponse performs getConfig() and serializes it to the handler's output
+func getConfigResponse(params admin_api.ConfigInfoParams) (*models.Configuration, error) {
+	mAdmin, err := newMAdminClient()
+	if err != nil {
+		log.Println("error creating Madmin Client:", err)
+		return nil, err
+	}
+	// create a MinIO Admin Client interface implementation
+	// defining the client to be used
+	adminClient := adminClient{client: mAdmin}
+
+	configTarget, err := getConfig(adminClient, params.Name)
+	if err != nil {
+		log.Println("error listing configurations:", err)
+		return nil, err
+	}
+
+	// serialize output
+	var confkv []*models.ConfigurationKV
+	confName := params.Name
+	for _, kv := range configTarget.KVS {
+		confkv = append(confkv, &models.ConfigurationKV{Key: kv.Key, Value: kv.Value})
+	}
```

Comment:
I think this should be inside the `getConfig` function since `getConfigResponse` should be only responsible to populate the values of `&models.Configuration` and not compute them